### PR TITLE
kernel: improve coverage for kernel/userspace.c

### PR DESCRIPTION
**Pull request by tyrantlizard77**
Brings us to 100% function and > 90% line coverage for this C file.

Found a bug while I was doing this, #17023 opened.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: tests/kernel/mem_protect/userspace/src/main.c [hunk 1]
```diff
@@ -1085,6 +1085,52 @@ void test_stack_buffer(void)
 
 }
 
+void z_impl_missing_syscall(void)
+{
+	/* Shouldn't ever get here; no handler function compiled */
+	k_panic();
```
Comment (ioannisg) [inline tests/kernel/mem_protect/userspace/src/main.c]: Could we remove this then?

File: tests/kernel/mem_protect/userspace/src/main.c [hunk 2]
```diff
@@ -1085,6 +1085,52 @@ void test_stack_buffer(void)
...
+}
+
+void test_unimplemented_syscall(void)
+{
+	expect_fault = true;
+	expected_reason = REASON_KERNEL_OOPS;
+
+	missing_syscall();
```
Comment (ioannisg) [inline tests/kernel/mem_protect/userspace/src/main.c]: How about when this is called from supervisor thread? We don't need a "handler" in that case, do we?

**Review by ioannisg — COMMENTED** [review]
Looks good to me, I left a couple of questions for clarifications.

File: tests/kernel/mem_protect/userspace/src/main.c (hunk 2 above)
Comment (ioannisg) [reply tests/kernel/mem_protect/userspace/src/main.c]: Also. should we test this scenario?

Comment (andrewboie) [reply tests/kernel/mem_protect/userspace/src/main.c]: no, we shouldn't.
the point of this is to show that when we invoke a missing system call from user mode, the appropriate fallback function runs (handler_no_syscall handler_bad_syscall)

**Review by ioannisg — APPROVED** [review]

File: tests/kernel/mem_protect/userspace/src/main.c (hunk 2 above)
Comment (ioannisg) [reply tests/kernel/mem_protect/userspace/src/main.c]: thanks

**Comment by ioannisg** [comment]
Userspace bad_syscall test fails on nRF9160 after this merge, @andrewboie

**Pull request merged**